src/main/kotlin/org/jetbrains/research/testspark/helpers/psi/kotlin/KotlinPsiClassWrapper.kt

```diff
@@ -0,0 +1,125 @@
+package org.jetbrains.research.testspark.helpers.psi.kotlin
+
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.psi.PsiFile
+import com.intellij.psi.search.GlobalSearchScope
+import com.intellij.psi.search.searches.ClassInheritorsSearch
+import com.intellij.psi.util.PsiTreeUtil
+import org.jetbrains.kotlin.asJava.toLightClass
+import org.jetbrains.kotlin.idea.caches.resolve.analyze
+import org.jetbrains.kotlin.idea.refactoring.isInterfaceClass
+import org.jetbrains.kotlin.lexer.KtTokens
+import org.jetbrains.kotlin.psi.KtClass
+import org.jetbrains.kotlin.psi.KtClassOrObject
+import org.jetbrains.kotlin.psi.KtObjectDeclaration
+import org.jetbrains.kotlin.resolve.BindingContext
+import org.jetbrains.kotlin.resolve.DescriptorToSourceUtils
+import org.jetbrains.research.testspark.core.data.ClassType
+import org.jetbrains.research.testspark.core.utils.importPattern
+import org.jetbrains.research.testspark.core.utils.packagePattern
+import org.jetbrains.research.testspark.helpers.psi.PsiClassWrapper
+import org.jetbrains.research.testspark.helpers.psi.PsiMethodWrapper
+
+class KotlinPsiClassWrapper(private val psiClass: KtClassOrObject) : PsiClassWrapper {
+    override val name: String get() = psiClass.name ?: ""
+
+    override val qualifiedName: String get() = psiClass.fqName?.asString() ?: ""
```

Comment:
Do we want to tolerate the PSI classes without `quialifiedName` and `name` properties? 

We create these `PsiClassWrapper` classes for PSI elements that cannot be tested, and afterward, somewhere-nobody-remembers-where, we filter out those `PsiClassWrapper`s whose `qualifiedName == ""`. 

It requires us to always remember the validation of `PsiClassWrapper` knowing that the underlying PSI element may not have a qualified name. 

Isn't it better to **fail fast** (`require(psiClass.qualifiedName !== null)` and have a try-catch block on the upper level?
```kotlin
try {
  val wrapper: PsiClassWrapper = KotlinPsiClassWrapper(psiClass);
} catch(err) {
  print(err);
}

// ...

class KotlinPsiClassWrapper(psiClass: PsiClass) {
   override val qualifiedName:String get() = psiClass.fqName!!.asString()
}
```

Or to make the `PsiClassWrapper` be created only for valid PSI elements, having an if-check whether the PSI element being considered has the `qualifiedName`, and only then create the `PsiClassWrapper`:
```kotlin
if (psiClass.hasQualifiedName()) {
  val wrapper: PsiClassWrapper = KotlinPsiClassWrapper(psiClass);
}

// ...

class KotlinPsiClassWrapper(psiClass: PsiClass) {
   override val qualifiedName: String get() = psiClass.fqName!!.asString()
}
```

What do you think? @Frosendroska @pderakhshanfar @arksap2002

Reply:
Hmm. I am not sure even in which scenario the qualified name is missing. Technically, the Qualified name is sth that we should always know. So, I agree with @Vladislav0Art that if we have a scenario in which we don't know the qualified name, there should probably be a bug, and it needs to be reflected in the software behavior as well.

Reply:
this is a good point, thank you! I like the solution with `psiClass.hasQualifiedName()` checking more.

Reply:
> Hmm. I am not sure even in which scenario the qualified name is missing. Technically, the Qualified name is sth that we should always know. So, I agree with @Vladislav0Art that if we have a scenario in which we don't know the qualified name, there should probably be a bug, and it needs to be reflected in the software behavior as well.

I guess the native Kotlin classes such as `String` and `Array` might have `qualifiedName == null`.